### ISSUE TITLE
Too many "Network is unreachable" messages consume space

### ISSUE THREAD

**Issue by codergem** [rsyslog/rsyslog#4740]
When a syslog server is unreachable, we see too many of the below error messages (proportional to the number of logs rsyslog gets to process). Since the UDPSend function in omfwd.c tries sending over > 1 sockets per every log, we see too many error messages in our log files. A configurable option to limit the number of this specific message would help. Though the error message is helpful, do we really need so many logs ?

Nov 24 12:09:30 MYHOST rsyslogd[30795]: omfwd/udp: socket 12: sendto() error: Network is unreachable [v8.1908.0 try https://www.rsyslog.com/e/2354 ]
Nov 24 12:09:30 MYHOST rsyslogd[30795]: omfwd/udp: socket 13: sendto() error: Network is unreachable [v8.1908.0 try https://www.rsyslog.com/e/2354 ]
Nov 24 12:09:30 MYHOST rsyslogd[30795]: omfwd: socket 13: error 101 sending via udp: Network is unreachable [v8.1908.0 try https://www.rsyslog.com/e/2354 ]

### Expected behavior
We see many error messages which rsyslog is unable to send, proportional to the number of logs rsyslog receives for processing.

### Actual behavior
We should see lesser number of these error log messages.

### Steps to reproduce the behavior
Configure an unreachable syslog server for UDP. Generate logs which rsyslog must attempt to send to unreachable server. Config used in our case:
action(type="omfwd" Target="10.1.1.1" Port="514" Protocol="udp" NetworkNamespace="some_VRF"   template="remoteFmt" Address="None" queue.type="LinkedList" queue.size="1000")

### Environment
- rsyslog version: v8.1908.0
- platform: linux

**Comment by z8k** [comment]
I would prefer to suppress the "Network is unreachable" messages and the related 
```
rsyslogd: action 'action-1-builtin:omfwd
' suspended (module 'builtin:omfwd'), retry 0. There should be messages before t
his one giving the reason for suspension. [v8.2302.0 try https://www.rsyslog.com
/e/2007 ]
```
messages that always follow it entirely. I chose UDP knowing there is no guarantee of delivery; that is fine. Many useless logs from rsyslog itself is annoying.

**Comment by rgerhards** [comment]
There should be a rate limiter for such messages. Maybe they occur just in bursts? Could probably be. That would not work will with the default ratelimit settings. I need to check a bit. Maybe just a config issue.

**Comment by z8k** [comment]
> Maybe they occur just in bursts?

Yes this is true, I see them around laptop suspend/resume events where the network goes away first and comes back last, and there is lots of activity in between.

**Comment by rgerhards** [comment]
I checked, and the default for the internal rate limiter (5 seconds) is very probably too low for your environment. 

See https://www.rsyslog.com/doc/master/rainerscript/global.html , on-page search for "internalmsg". You'll find the params that you can adjust to get rid of the excess messages.

**Comment by z8k** [comment]
Thanks for the hint. I have added filters to suppress the messages I'm not interested in, and now my logs are tidy again.
